### PR TITLE
Handle GreaterThan constraints

### LINKED ISSUE #14: Handle GreaterThan constraints
can be easily cast to LessThan constraints - at least for current QPTH method

### PR DESCRIPTION
**Pull request by AKS1996**
closes #14

### CODE REVIEW

**Comment by AKS1996** [comment]
we'll need to handle `Nonpositive` cones in MOSD too

File: test/MOI_wrapper.jl [hunk 1]
```diff
@@ -499,6 +499,35 @@ end
...
+
+    # define objective
+    objective_function = MOI.ScalarAffineFunction(MOI.ScalarAffineTerm.([1.0], x), 0.0)
+    MOI.set(model, MOI.ObjectiveFunction{MOI.ScalarAffineFunction{Float64}}(), objective_function)
+    MOI.set(model, MOI.ObjectiveSense(), MOI.MIN_SENSE)
+
+    MOI.add_constraint(
+        model,
+        MOI.ScalarAffineFunction(MOI.ScalarAffineTerm.([1.0], x), 0.),
+        MOI.GreaterThan(3.0)
```
Comment (matbesancon) [inline test/MOI_wrapper.jl:519]:
```suggestion
        MOI.GreaterThan(3.0),
```

**Comment by matbesancon** [comment]
We should probably have a bunch more tests with SAF in <=, >= and SV in both too to avoid bad surprises later

File: test/MOI_wrapper.jl [hunk 2]
```diff
@@ -694,6 +723,102 @@ end
...
+
+@testset "Differentiating LP with SAF, SV with LE, GE constraints" begin
+    """
+        max 2x + 3y + 4z
+        s.t. 3x+2y+z <= 10
+            2x+5y+3z <= 15
+            x ≤ 3
+            0 ≤ y ≤ 2
+            6 ≥ z ≥ -1
+            x,y,z >= 0
```
Comment (matbesancon) [inline test/MOI_wrapper.jl:735]:
```suggestion
        max 2x + 3y + 4z
        s.t. 3x+2y+z <= 10
              2x+5y+3z <= 15
              x ≤ 3
              0 ≤ y ≤ 2
              6 ≥ z ≥ -1
              x, y, z >= 0
```

File: test/MOI_wrapper.jl [hunk 3]
```diff
@@ -694,6 +723,102 @@ end
...
+
+@testset "Differentiating LP with SAF, SV with LE, GE constraints" begin
+    """
+        max 2x + 3y + 4z
+        s.t. 3x+2y+z <= 10
+              2x+5y+3z <= 15
+              x ≤ 3
+              0 ≤ y ≤ 2
+              6 ≥ z ≥ -1
+              x, y, z >= 0
```
Comment (matbesancon) [inline test/MOI_wrapper.jl:735]:
```suggestion
        max  2x + 3y + 4z
        s.t. 3x+2y+z <= 10
             2x+5y+3z <= 15
             x ≤ 3
             0 ≤ y ≤ 2
             6 ≥ z ≥ -1
             x, y, z >= 0
```

**Review by matbesancon — APPROVED** [review]

File: src/MOI_wrapper.jl [hunk 4]
```diff
@@ -352,16 +305,28 @@ function backward_quad(model::Optimizer, params::Vector{String}, dl_dz::Vector{F
...
             neq_sv, eq_con_sv_idx,
         ) = problem_data
 
         # separate λ, ν
 
-        λ = MOI.get.(model.optimizer, MOI.ConstraintDual(), ineq_con_idx)
+        λ = MOI.get.(model.optimizer, MOI.ConstraintDual(), le_con_idx)
+        append!(
+            λ,
+            MOI.get.(model.optimizer, MOI.ConstraintDual(), ge_con_idx)
```
Comment (matbesancon) [inline src/MOI_wrapper.jl:321]:
```suggestion
            MOI.get.(model.optimizer, MOI.ConstraintDual(), ge_con_idx),
```

File: src/MOI_wrapper.jl [hunk 5]
```diff
@@ -352,16 +305,28 @@ function backward_quad(model::Optimizer, params::Vector{String}, dl_dz::Vector{F
...
+        )
+        append!(
+            λ,
+            MOI.get.(model.optimizer, MOI.ConstraintDual(), le_con_sv_idx)
```
Comment (matbesancon) [inline src/MOI_wrapper.jl:325]:
```suggestion
            MOI.get.(model.optimizer, MOI.ConstraintDual(), le_con_sv_idx),
```

**Comment by joaquimg** [comment]
Anything stopping the merge?

**Comment by matbesancon** [comment]
nope all good!

**Comment by joaquimg** [comment]
Nice!

**Pull request merged**